rules_building_block/discovery_generic_registry_query.toml

```diff
@@ -14,8 +14,9 @@ description = """
 This rule identifies the execution of commands that can be used to query the Windows Registry. Adversaries may query the
 registry to gain situational awareness about the host, like installed security software, programs and settings.
 """
-from = "now-9m"
-index = ["logs-endpoint.events.*"]
+from = "now-24h"
+index = ["logs-endpoint.events.process*"]
+interval = "24h"
```

Comment:
Process specific index adjustment makes sense. However, I am not sure of the trade-off with 24h lookback with a 14 day history window as well. While we reduce the frequency of this run, we potentially increase resources used as well to hold such data. CPU wise, yes we would reduce a re-occurring job but may increase exponentially when the job runs.

If a customer is ingesting 3-5 TB a day, for example, and we have a query that attempts to historically check for a new occurrence in the last 14 days, that could be quite expensive. What if the history time window was simply reduced to 7 days? 

Also if this is a BBR rule and we want other rules to rely on it for detection, then we may be introducing delayed alerting for such events.

Reply:
Current implementation:
every 5 mins, look back 9m for a new occurrence over the last 14 days

This change:
every 24h, look back 24h for a new occurrence over the last 14 days

So really, the 14 days stays constant and will not have a big impact. The bigger impact will be 24h vs 9m, but I think it is negligible enough since it is a single run per day. Most new terms history windows tend to be larger by design.